sucury.py

```diff
@@ -173,27 +173,43 @@ def update(self):
             self.head.y += self.ymov * GRID_SIZE
 
 ##
-## The apple class.
+## The fruit class.
 ##
 
-class Apple:
+class Fruit:
     def __init__(self):
 
         # Pick a random position within the game arena
         self.x = int(random.randint(0, WIDTH)/GRID_SIZE) * GRID_SIZE
         self.y = int(random.randint(0, HEIGHT)/GRID_SIZE) * GRID_SIZE
 
-        # Create an apple at that location
+        # Create an fruit at that location
         self.rect = pygame.Rect(self.x, self.y, GRID_SIZE, GRID_SIZE)
 
+        self.type = random.randint(0, 3)
+
```

Comment:
in the implementation of the Fruit class, the position of the fruit is randomly generated. however, there's no check to ensure that this randomly chosen position does not overlap with the current location of the snake. this can lead to scenarios where the fruit appears directly on the snake, which may not be desirable for gameplay.
to address this issue, I suggest modifying the Fruit class constructor to ensure that the fruit is not placed on any part of the snake (either the head or the tail segments). 


```
class Fruit:
    def __init__(self, snake):
        self.rect = pygame.Rect(0, 0, GRID_SIZE, GRID_SIZE)
        self.place_fruit(snake)
    def place_fruit(self, snake):
        while True:
            # Randomly select a position within the game screen boundaries
            self.x = random.randint(0, (WIDTH - GRID_SIZE) // GRID_SIZE) * GRID_SIZE
            self.y = random.randint(0, (HEIGHT - GRID_SIZE) // GRID_SIZE) * GRID_SIZE
            self.rect.topleft = (self.x, self.y)
            # Check if the position overlaps with the snake
            if self.rect.collidelist([snake.head] + snake.tail) == -1:
                break  # The position is valid, exit the loop
```